docs/non_smooth.rst

```diff
@@ -22,12 +22,69 @@ Proximal gradient
 
     jaxopt.ProximalGradient
 
+Instantiating and running the solver
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Proximal gradient is a generalization of :ref:`projected gradient descent
+<constrained_optim>`. The non-smooth term :math:`g` above is specified by
+setting the corresponding ``prox`` argument.
+
+For instance, suppose we want to minimize
+
+.. math::
+
+    \min_{w} \frac{1}{2n} ||Xw - y||^2 + \lambda ||w||_1
```

Comment:
Ideally this would be an argmin, but I don't know if it's easy to do with sphinx